core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -58,167 +57,248 @@ public Sep10Service(
+    validateChallengeRequestClient(request);
+    // Validate the validity of the memo
+    Memo memo = validateChallengeRequestMemo(request);
+    // post validations to be defined by the anchor
+    postChallengeRequestValidation(request);
+    // increment counter
+    incrementChallengeRequestCreatedCounter();
+    // Create the challenge
+    return createChallengeResponse(request, memo);
+  }
+
+  public ValidationResponse validateChallenge(ValidationRequest request)
+      throws IOException, InvalidSep10ChallengeException, SepValidationException {
+    info("Validating SEP-10 challenge.");
+
+    ChallengeTransaction challenge = parseChallenge(request);
+    // pre validation to be defined by the anchor
+    preValidateRequestValidation(request, challenge);
+    // fetch the client domain from the transaction
+    String clientDomain = fetchClientDomain(challenge);
+    // fetch the account response from the horizon
+    AccountResponse account = fetchAccount(request, challenge, clientDomain);
+
+    if (account == null) {
+      // The account does not exist from Horizon, using the client's master key to verify.
+      return ValidationResponse.of(generateSep10Jwt(challenge, clientDomain));
+    }
+    // Since the account exists, we should check the signers and the client domain
+    validateChallengeRequest(request, account, clientDomain);
+    // increment counter
+    incrementValidationRequestValidatedCounter();
+    // Generate the JWT token
+    return ValidationResponse.of(generateSep10Jwt(challenge, clientDomain));
+  }
+
+  @Override
+  public ChallengeResponse createChallengeResponse(ChallengeRequest request, Memo memo)
+      throws SepException {
+    try {
+      String clientSigningKey = null;
+      if (!Objects.toString(request.getClientDomain(), "").isEmpty()) {
+        debugF("Fetching SIGNING_KEY from client_domain: {}", request.getClientDomain());
+        clientSigningKey = fetchSigningKeyFromClientDomain(request.getClientDomain());
+        debugF("SIGNING_KEY from client_domain fetched: {}", clientSigningKey);
+      }
+
+      // create challenge transaction
+      Transaction txn = newChallenge(request, clientSigningKey, memo);
+
+      // Convert the challenge to response
+      trace("SEP-10 challenge txn:", txn);
+      ChallengeResponse challengeResponse =
+          ChallengeResponse.of(txn.toEnvelopeXdrBase64(), appConfig.getStellarNetworkPassphrase());
+      trace("challengeResponse:", challengeResponse);
+      return challengeResponse;
+    } catch (InvalidSep10ChallengeException ex) {
+      warnEx(ex);
+      throw new SepException("Failed to create the sep-10 challenge.", ex);
+    }
+  }
+
+  Transaction newChallenge(ChallengeRequest request, String clientSigningKey, Memo memo)
+      throws InvalidSep10ChallengeException {
+
+    KeyPair signer = KeyPair.fromSecretSeed(secretConfig.getSep10SigningSeed());
+    long now = System.currentTimeMillis() / 1000L;
```

Comment:
We should probably avoid calling `System#currentTimeMillis` and `Instant#now` and replace them with a `Clock` call, so that we can test time-related code.

Reply:
Replace with `Instant.now().getEpochSecond()`.